src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj

```diff
@@ -35,6 +35,7 @@
     <PackageReference Include="System.Net.Primitives" Version="$(SystemNetPrimitivesVersion)" />
     <PackageReference Include="System.Collections" Version="$(SystemCollectionsVersion)" />
     <PackageReference Include="System.Runtime.InteropServices" Version="$(SystemRuntimeInteropServicesVersion)" />
+    <PackageReference Include="System.Text.Encodings.Web" Version="4.5.1" />
```

Comment:
I don't think this is the "correct" fix. This is just fixing the symptom, but not the underlying issue here.

The underlying issue is that Unity and Lamar implementations are referencing the old version. Any customer using them are going to have the same problem.

We should instead get them updated to no longer reference the old version.

The reason they reference the old version is because they both reference https://www.nuget.org/packages/Microsoft.AspNetCore.Hosting.Abstractions/, which references https://www.nuget.org/packages/Microsoft.AspNetCore.Http.Abstractions/. According to @davidfowl, both of these packages are EOL. So I'm not sure why they are still getting referenced by the latest Unity and Lamar implementations.

I think we should get those dependencies fixed, and then we update to the latest versions. That will fix this issue for us, and for our customers.

Reply:
Thanks @eerhardt for the feedback. I'm not fully familiar with the nuget package dependency graph resolutions and policies and appreciate your thoughts.

1. Regarding `Microsoft.AspNetCore.*.Abstractions` packages comment you made above, is the case that these packages are no longer supported or you just mean the version that was used by Unity and Lamar? If the latter, then the fix would be to use the latest version of the Abstractions packages (perhaps by using a compatible later version of lamar and unity)? If the former, I'm a little puzzled since the MS abstractions packages seem to be very widely used and I would expect a more centralized migration strategy.
2. Regarding getting fixes for the lamar and unity packages, at least the unity package seems [not be active](https://github.com/unitycontainer/microsoft-dependency-injection) any more. What is the guidance for consumers of EOL packages like above in this case?

Reply:
If the project isn't under active development any more (which it doesn't look like it is), I don't think we should be looking to them for a fix.

I think we have two choices: drop the tests that use this package, or fix it for ourselves.

Reply:
Also, looks like https://www.nuget.org/packages/Microsoft.AspNetCore.Hosting.Abstractions/ is still listed and doesn't have any updates. If that package is EOL, what is the policy?

Reply:
> the unity package seems [not be active](https://github.com/unitycontainer/microsoft-dependency-injection) any more. What is the guidance for consumers of EOL packages like above in this case?

I opened #64988 to remove Unity and StructureMap from our tests since they are both EOL.

> Also, looks like https://www.nuget.org/packages/Microsoft.AspNetCore.Hosting.Abstractions/ is still listed and doesn't have any updates. If that package is EOL, what is the policy?

@mkArtakMSFT @halter73 @pranavkm - Can you answer this question? Is Microsoft.AspNetCore.Hosting.Abstractions EOL? And if it is, should we be telling our customers like the [Lamar DI](https://www.nuget.org/packages/Lamar.Microsoft.DependencyInjection/) to not be using it?

Reply:
The 2.2 version of the package is EOL and we're working on getting it unlisted / obsoleted on NuGet.org. The 2.1.x series (e.g. https://www.nuget.org/packages/Microsoft.AspNetCore.Hosting.Abstractions/2.1.1) is perpetually supported since it's the last LTS version that works on .NET Framework.

Reply:
> And if it is, should we be telling our customers like the [Lamar DI](https://www.nuget.org/packages/Lamar.Microsoft.DependencyInjection/) to not be using it?

Yup. They should be using the ASP.NET Core shared framework instead: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/target-aspnetcore?view=aspnetcore-6.0&tabs=visual-studio

Reply:
I've opened https://github.com/JasperFx/lamar/issues/328 for Lamar DI to remove its dependency on this EOL package.